test/unittest/dds/subscriber/DataReaderTests.cpp

```diff
@@ -845,6 +845,218 @@ TEST_F(DataReaderTests, resource_limits)
     }
 }
 
+void check_sample(
+        const SampleInfo& info_1,
+        const SampleInfo& info_2,
+        const FooType& data_1,
+        const FooType& data_2,
+        bool should_be_equal)
+{
+    EXPECT_EQ(should_be_equal, info_1.sample_identity == info_2.sample_identity);
+    EXPECT_EQ(should_be_equal, data_1 == data_2);
```

Comment:
It looks like field `message_` is an array with length depending on the data given, so it could be not fully initialized.
Thus, this comprobation should be just checking `message_` are equal in their length and not the whole array (or preinitialize them completly).

Reply:
I finally realized it is enough (for this test) to just check the contents (as done by `check_sample_values`), so I removed the other checks and check_sample method on 898ed3e